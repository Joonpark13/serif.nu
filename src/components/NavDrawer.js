import React from 'react';
import PropTypes from 'prop-types';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import { Link } from 'react-router-dom';

/* istanbul ignore next */
function ListItemLink(props) {
  return <ListItem button component={Link} {...props} />;
}

function NavDrawer({ isOpen, closeFunc }) {
  return (
    <Drawer onClose={closeFunc} open={isOpen}>
      <List>
        <ListItemLink to="/" onClick={closeFunc}>
          <ListItemText primary="Serif.nu" />
        </ListItemLink>
        <ListItemLink to="/about" onClick={closeFunc}>
          <ListItemText primary="About" />
        </ListItemLink>
        <ListItemLink to="/faq" onClick={closeFunc}>
          <ListItemText primary="FAQ" />
        </ListItemLink>
        <ListItemLink to="/report" onClick={closeFunc}>
          <ListItemText primary="Report a Bug" />
        </ListItemLink>
        <ListItemLink to="/contact" onClick={closeFunc}>
          <ListItemText primary="Contact" />
        </ListItemLink>
      </List>
    </Drawer>
  );
}

NavDrawer.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  closeFunc: PropTypes.func.isRequired,
};

export default NavDrawer;
