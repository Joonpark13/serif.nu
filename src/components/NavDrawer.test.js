import React from 'react';
import { shallow } from 'enzyme';
import NavDrawer from './NavDrawer';

describe('NavDrawer', () => {
  const defaultProps = {
    isOpen: true,
    closeFunc: () => {},
  };

  it('should render correctly', () => {
    const wrapper = shallow(<NavDrawer {...defaultProps} />);

    expect(wrapper.get(0)).toMatchSnapshot();
  });
});
