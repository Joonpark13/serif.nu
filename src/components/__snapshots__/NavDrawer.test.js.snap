// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDrawer should render correctly 1`] = `
<ForwardRef(WithStyles)
  onClose={[Function]}
  open={true}
>
  <div
    className=""
  >
    <ForwardRef(WithStyles)>
      <ListItemLink
        onClick={[Function]}
        to="/"
      >
        <ForwardRef(WithStyles)
          primary="Serif.nu"
        />
      </ListItemLink>
      <ListItemLink
        onClick={[Function]}
        to="/about"
      >
        <ForwardRef(WithStyles)
          primary="About"
        />
      </ListItemLink>
      <ListItemLink
        onClick={[Function]}
        to="/faq"
      >
        <ForwardRef(WithStyles)
          primary="FAQ"
        />
      </ListItemLink>
      <ListItemLink
        onClick={[Function]}
        to="/bugs"
      >
        <ForwardRef(WithStyles)
          primary="Report a Bug"
        />
      </ListItemLink>
      <ListItemLink
        onClick={[Function]}
        to="/contact"
      >
        <ForwardRef(WithStyles)
          primary="Contact"
        />
      </ListItemLink>
    </ForwardRef(WithStyles)>
    <div
      className=""
    >
      <Link
        onClick={[Function]}
        to="/legal"
      >
        <ForwardRef(WithStyles)
          variant="caption"
        >
          Legal notices
        </ForwardRef(WithStyles)>
      </Link>
    </div>
  </div>
</ForwardRef(WithStyles)>
`;
